ext/include/opentelemetry/ext/zpages/zpages_http_server.h

```diff
@@ -0,0 +1,188 @@
+#pragma once
+
+#include <algorithm>
+#include <fstream>
+#include <iostream>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include "opentelemetry/ext/http/server/HttpServer.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace ext {
+namespace zpages {
+
+class zPagesHttpServer : public HTTP_SERVER_NS::HttpServer {
+ protected:
+  /*
+   * Construct the server by initializing the endpoint for serving static files, which show up on the
+   * web if the user is on the given host:port. Static files can be seen relative to the folder where the
+   * executable was ran.
+   * @param host is the host where the TraceZ webpages will be displayed
+   * @param port is the port where the TraceZ webpages will be displayed
+   * @param endpoint is where this specific zPage will server files
+   */
+  zPagesHttpServer(const std::string& endpoint, std::string host, int port) : HttpServer(), endpoint_(endpoint) {
+    std::ostringstream os;
+    os << host << ":" << port;
+    setServerName(os.str());
+    addListeningPort(port);
+  };
+
+  /*
+   * Set the HTTP server to serve static files from the root of host:port
+   * @param server should be an instance of this object
+   */
+  void InitializeFileEndpoint(zPagesHttpServer& server) {
+    server[root_endpt_] = ServeFile;
+  }
+  
+    /*
+   * Helper function tat returns query information by isolating it from the base endpoint
+   */
+  std::string GetQuery(const std::string& uri) {
+    if (endpoint_.length() + 1 > uri.length()) return uri;
+    return uri.substr(endpoint_.length() + 1);
+  }
+
+  /*
+   * Helper that returns whether a str starts with pre
+   * @param str is the string we're checking
+   * @param pre is the prefix we're checking against
+   */
+  bool StartsWith(const std::string& str, std::string pre) {
+    return (pre.length() > str.length())
+        ? false
+        : str.substr(0, pre.length()) == pre;
```

Comment:
Very clean, thank you!

Reply:
Also, would that be acceptable as a 1-line function? I feel like it would be, since just that line of code could be confusing? On the other hand, I think we could comment what that line does if we replace the function instead of making a whole new function?

Reply:
Yes, I think in this case it is ok to keep it as one-liner, because it is used in multiple places and StartsWith is more explicitly stating what we want to check.